scene/gui/option_button.cpp

```diff
@@ -504,6 +504,13 @@ void OptionButton::show_popup() {
 
 	Rect2 rect = get_screen_rect();
 	rect.position.y += rect.size.height;
+	if (get_icon().is_valid() && is_expand_icon()) {
+		Size2 _size = get_size() - theme_cache.normal->get_offset() * 2;
+		float icon_width = get_icon()->get_width() * _size.height / get_icon()->get_height();
```

Comment:
I don't think this logic is entirely correct. Button's own icon max size, for example, is applied regardless of the expand mode. This code seems to replicate a small part of the size resolution logic from `button.cpp`'s `NOTIFICATION_DRAW` handler.

I think we should instead extract that part of the Button's code into a dedicated method and use that method here as well.

Reply:
Yes, I extracted the way the icons were resized when rendered on the `NOTIFICATION_DRAW` portion. I've implemented your suggestion by moving that math into functions (`get_icon_width_with_aspect_ratio(...)`) and refactored to use these new methods.

Sorry for the delay, I've been busy the last month, I only had some free time to look at this again now.